primitives/manta/src/xcm.rs

```diff
@@ -69,21 +71,27 @@ impl Reserve for MultiAsset {
     }
 }
 
-/// A `FilterAssetLocation` implementation. Filters multi native assets whose
-/// reserve is same with `origin`.
+/// Multi-Native Asset Filter
+///
+/// Filters multi-native assets whose reserve is same as the `origin`.
 pub struct MultiNativeAsset;
+
 impl FilterAssetLocation for MultiNativeAsset {
+    #[inline]
     fn filter_asset_location(asset: &MultiAsset, origin: &MultiLocation) -> bool {
         asset.reserve().map(|r| r == *origin).unwrap_or(false)
     }
 }
 
-pub struct AccountIdToMultiLocation<AccountId>(PhantomData<AccountId>);
-impl<AccountId> Convert<AccountId, MultiLocation> for AccountIdToMultiLocation<AccountId>
+/// AccountId to [`MultiLocation`] Converter
+pub struct AccountIdToMultiLocation;
+
+impl<A> Convert<A, MultiLocation> for AccountIdToMultiLocation
```

Comment:
the generics changes are great, but they make the PR big and hard to reason about.
For the future please try keeping a single-responsibility-principle in PRs where there is only one functional change per PR that is not conflated with other changes.
If there's a large many-files refactoring, it's well worth to make that its own PR.

Reply:
yeah totally, I think Brandon and I were rushing again : (